packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -189,52 +271,81 @@ export const prepareStakingAccountKit = (baggage, makeRecorderKit, zcf) => {
         /**
          * _Assumes users has already sent funds to their ICA, until #9193
          * @param {CosmosValidatorAddress} validator
-         * @param {Amount<'nat'>} ertpAmount
+         * @param {AmountArg} amount
          */
-        async delegate(validator, ertpAmount) {
-          trace('delegate', validator, ertpAmount);
-
-          // FIXME brand handling and amount scaling #9211
-          trace('TODO: handle brand', ertpAmount);
-          const amount = {
-            amount: String(ertpAmount.value),
-            denom: this.state.bondDenom,
-          };
-
-          const account = this.facets.helper.owned();
-          const delegatorAddress = this.state.chainAddress.address;
+        async delegate(validator, amount) {
+          trace('delegate', validator, amount);
+          const { helper } = this.facets;
+          const { chainAddress } = this.state;
 
-          const result = await E(account).executeEncodedTx([
+          // FIXME get values from proposal or args
+          const result = await E(helper.owned()).executeEncodedTx([
             toAnyJSON(
               MsgDelegate.toProtoMsg({
-                delegatorAddress,
+                delegatorAddress: chainAddress.address,
                 validatorAddress: validator.address,
-                amount,
+                amount: helper.amountToCoin(amount),
               }),
             ),
           ]);
 
-          if (!result) throw Fail`Failed to delegate.`;
-          return tryDecodeResponse(result, MsgDelegateResponse.fromProtoMsg);
+          const response = tryDecodeResponse(
+            result,
+            MsgDelegateResponse.fromProtoMsg,
+          );
+          mustMatch(harden(response), harden({})); // TODO: ignore response??
+        },
+        /**
+         * _Assumes users has already sent funds to their ICA, until #9193
+         * @param {CosmosValidatorAddress} srcValidator
+         * @param {CosmosValidatorAddress} dstValidator
+         * @param {AmountArg} amount
+         */
+        async redelegate(srcValidator, dstValidator, amount) {
+          trace('redelegate', srcValidator, dstValidator, amount);
+          const { helper } = this.facets;
+          const { chainAddress } = this.state;
+
+          // FIXME get values from proposal or args
+          const result = await E(helper.owned()).executeEncodedTx([
+            toAnyJSON(
+              MsgBeginRedelegate.toProtoMsg({
+                delegatorAddress: chainAddress.address,
+                validatorSrcAddress: srcValidator.address,
+                validatorDstAddress: dstValidator.address,
+                amount: helper.amountToCoin(amount),
+              }),
+            ),
+          ]);
+
+          const response = tryDecodeResponse(
+            result,
+            MsgBeginRedelegateResponse.fromProtoMsg,
+          );
+          trace('MsgBeginRedelegateResponse', response);
+          // TODO? return completionTime?
```

Comment:
The spec says to return `void`. Do we really want to throw away `completionTime`?
(product question)

Reply:
Since this is a _Cosmos_ staking kit, I think we can return the Cosmos type. The higher level APIs can be more abstract. They might want to return void or a TimeRecord.

Reply:
oh. based on discussion above, I just re-worked it to ignore the return value altogether.
I hope that's ok.